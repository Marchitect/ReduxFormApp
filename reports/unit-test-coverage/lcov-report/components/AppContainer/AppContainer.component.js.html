
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for components/AppContainer/AppContainer.component.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../index.html">All files</a> / <a href="index.html">components/AppContainer</a> AppContainer.component.js</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">50% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>2/4</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>0/0</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">33.33% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>1/3</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">50% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>2/4</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
    </div>
    <div class='status-line medium'></div>
    <pre><table class="coverage">
<tr><td class="line-count quiet"><a name='L1'></a><a href='#L1'>1</a>
<a name='L2'></a><a href='#L2'>2</a>
<a name='L3'></a><a href='#L3'>3</a>
<a name='L4'></a><a href='#L4'>4</a>
<a name='L5'></a><a href='#L5'>5</a>
<a name='L6'></a><a href='#L6'>6</a>
<a name='L7'></a><a href='#L7'>7</a>
<a name='L8'></a><a href='#L8'>8</a>
<a name='L9'></a><a href='#L9'>9</a>
<a name='L10'></a><a href='#L10'>10</a>
<a name='L11'></a><a href='#L11'>11</a>
<a name='L12'></a><a href='#L12'>12</a>
<a name='L13'></a><a href='#L13'>13</a>
<a name='L14'></a><a href='#L14'>14</a>
<a name='L15'></a><a href='#L15'>15</a>
<a name='L16'></a><a href='#L16'>16</a>
<a name='L17'></a><a href='#L17'>17</a>
<a name='L18'></a><a href='#L18'>18</a>
<a name='L19'></a><a href='#L19'>19</a>
<a name='L20'></a><a href='#L20'>20</a>
<a name='L21'></a><a href='#L21'>21</a></td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { useState } from "react"
import { Button } from "react-bootstrap"
import { FormModal } from "../FormModal"
import { DisplayTable } from "../DisplayTable"
import { FlexContainer } from "./AppContainer.style"
&nbsp;
export function AppContainer() {
  // state hook returns the current state and the function which updates the state
  const [modalShow, setModalShow] = React.useState(false)
&nbsp;
  return (
    &lt;FlexContainer&gt;
      &lt;Button variant="primary" onClick={<span class="fstat-no" title="function not covered" >()</span> =&gt; <span class="cstat-no" title="statement not covered" >setModalShow(true)}</span>&gt;
        Add new employee
      &lt;/Button&gt;
      &lt;DisplayTable /&gt;
      &lt;FormModal show={modalShow} onHide={<span class="fstat-no" title="function not covered" >()</span> =&gt; <span class="cstat-no" title="statement not covered" >setModalShow(false)}</span> /&gt;
    &lt;/FlexContainer&gt;
  )
}
&nbsp;</pre></td></tr></table></pre>

                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank">istanbul</a>
                at Mon Mar 16 2020 10:53:20 GMT+1100 (Australian Eastern Daylight Time)
            </div>
        </div>
        <script src="../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../sorter.js"></script>
        <script src="../../block-navigation.js"></script>
    </body>
</html>
    