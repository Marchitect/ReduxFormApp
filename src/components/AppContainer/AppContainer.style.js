// AppContainer.style.js
import styled from "styled-components"

export const FlexContainer = styled.div`
  font-size: 16px;
  display: flex;
  align-items: center;
  flex-direction: column;
  justify-content: space-around;
`
