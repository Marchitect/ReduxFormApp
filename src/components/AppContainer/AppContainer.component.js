import React, { useState } from "react"
import { Button } from "react-bootstrap"
import { FormModal } from "../FormModal"
import { DisplayTable } from "../DisplayTable"
import { FlexContainer } from "./AppContainer.style"

export function AppContainer() {
  // state hook returns the current state and the function which updates the state
  const [modalShow, setModalShow] = React.useState(false)

  return (
    <FlexContainer>
      <Button variant="primary" onClick={() => setModalShow(true)}>
        Add new employee
      </Button>
      <DisplayTable />
      <FormModal show={modalShow} onHide={() => setModalShow(false)} />
    </FlexContainer>
  )
}
