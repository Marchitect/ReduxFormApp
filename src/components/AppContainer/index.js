// index.js
import { AppContainer } from "./AppContainer.component"

export { AppContainer }
