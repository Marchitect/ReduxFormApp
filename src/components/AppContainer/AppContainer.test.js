// AppContainer.test.js
import React from "react"
import { AppContainer } from "./AppContainer.component"
import { render } from "@testing-library/react"

it("AppContainer renders correctly", () => {
  const appContainer = render(
      <AppContainer
        
      >
      </AppContainer>
    )
  expect(appContainer).toMatchSnapshot()
})