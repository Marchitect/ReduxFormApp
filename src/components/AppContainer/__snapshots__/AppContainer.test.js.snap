// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppContainer renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-AxmLO dngniA"
      >
        <div>
          <form
            class="sc-AxhCb hLMWUN"
          >
            <label
              class="sc-AxhUy kbqZXj"
              for="todo-item-input"
            >
              Enter your to-do item here:
            </label>
            <div
              class="sc-AxjAm iaLYws"
            >
              <input
                class="sc-AxirZ cvZsoc"
                name="todo-item-input"
                type="text"
                value=""
              />
              <button
                class="sc-AxiKw glrngo"
              >
                Submit
              </button>
            </div>
          </form>
        </div>
        <div
          class="sc-Axmtr ebnLPY"
        >
          <div
            class="sc-AxgMl zcOjk"
          >
            Learn React state hook
          </div>
          <button
            class="sc-AxheI eqXGvy"
          >
            Complete
          </button>
        </div>
        <div
          class="sc-Axmtr ebnLPY"
        >
          <div
            class="sc-AxgMl zcOjk"
          >
            Meet friends for lunch
          </div>
          <button
            class="sc-AxheI eqXGvy"
          >
            Complete
          </button>
        </div>
        <div
          class="sc-Axmtr ebnLPY"
        >
          <div
            class="sc-AxgMl zcOjk"
          >
            Learn React effect hook
          </div>
          <button
            class="sc-AxheI eqXGvy"
          >
            Complete
          </button>
        </div>
        <div
          class="sc-Axmtr ebnLPY"
        >
          <div
            class="sc-AxgMl zcOjk"
          >
            Buy groceries
          </div>
          <button
            class="sc-AxheI eqXGvy"
          >
            Complete
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-AxmLO dngniA"
    >
      <div>
        <form
          class="sc-AxhCb hLMWUN"
        >
          <label
            class="sc-AxhUy kbqZXj"
            for="todo-item-input"
          >
            Enter your to-do item here:
          </label>
          <div
            class="sc-AxjAm iaLYws"
          >
            <input
              class="sc-AxirZ cvZsoc"
              name="todo-item-input"
              type="text"
              value=""
            />
            <button
              class="sc-AxiKw glrngo"
            >
              Submit
            </button>
          </div>
        </form>
      </div>
      <div
        class="sc-Axmtr ebnLPY"
      >
        <div
          class="sc-AxgMl zcOjk"
        >
          Learn React state hook
        </div>
        <button
          class="sc-AxheI eqXGvy"
        >
          Complete
        </button>
      </div>
      <div
        class="sc-Axmtr ebnLPY"
      >
        <div
          class="sc-AxgMl zcOjk"
        >
          Meet friends for lunch
        </div>
        <button
          class="sc-AxheI eqXGvy"
        >
          Complete
        </button>
      </div>
      <div
        class="sc-Axmtr ebnLPY"
      >
        <div
          class="sc-AxgMl zcOjk"
        >
          Learn React effect hook
        </div>
        <button
          class="sc-AxheI eqXGvy"
        >
          Complete
        </button>
      </div>
      <div
        class="sc-Axmtr ebnLPY"
      >
        <div
          class="sc-AxgMl zcOjk"
        >
          Buy groceries
        </div>
        <button
          class="sc-AxheI eqXGvy"
        >
          Complete
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
